src/index.js

```diff
@@ -194,8 +196,52 @@ const HMMediaWeightSidebar = () => {
 		};
 	} );
 
+	const insertIframe = ( url, width = '1600px', height = '800px' ) => {
+		const iframeWindow = document.getElementById( 'post-preview-iframe' );
+
+		if ( iframeWindow ) {
+			// Reload the iframe to recalculate performance entries.
+			iframeWindow.contentWindow.location.reload();
+		} else {
+			const iframe = document.createElement( 'iframe' );
+			iframe.src = url;
+			iframe.width = width;
+			iframe.height = height;
+			iframe.id = 'post-preview-iframe';
+			iframe.style.display = 'none';
+			document.body.appendChild( iframe );
+		}
+	}
+
+	const previewPostLink = useSelect( ( select ) => select( 'core/editor' ).getEditedPostPreviewLink() );
+	const iframeURL = addQueryArgs( previewPostLink, { mediaWeight: '1' } );
+
+	useEffect( () => {
+		const listener = window.addEventListener( 'message', ( event ) => {
+			const receivedEntries = event.data;
+			// eslint-disable-next-line no-console
+			console.log( receivedEntries );
+			// You can call a useState callback here to get the data into the component.
+		} );
+
+		return () => {
+			window.removeEventListener( listener );
+		};
```

Comment:
This causes an error, `removeEventListener` needs the type specified; and `addEventListener` does not return anything, so this isn't doing what you'd expect. You'd want something more like this:

```js
const listener = ( event ) => { /* ... */ };
window.addEventListener( 'message', listener );
return () => { window.removeEventListener( 'message', listener ); };
```
that way you're unbinding the listener by reference.

```suggestion
		const listener = ( event ) => {
			const receivedEntries = event.data;
			// eslint-disable-next-line no-console
			console.log( receivedEntries );
			// You can call a useState callback here to get the data into the component.
		};
		window.addEventListener( 'message', listener );

		return () => {
			window.removeEventListener( listener );
		};
```
☝️ Untested but should work

Reply:
It looks like it should work, but `removeEventListener` needs the event name.

```
window.removeEventListener( 'message', listener );
```
:trollface:

Reply:
I updated the listener as suggested.